airbyte-integrations/connectors/destination-s3/src/main/java/io/airbyte/integrations/destination/s3/S3DestinationStrictEncrypt.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.s3;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.annotations.VisibleForTesting;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.AirbyteConnectionStatus.Status;
+
+public class S3DestinationStrictEncrypt extends S3Destination {
+
+  public S3DestinationStrictEncrypt() {
+    super();
+  }
+
+  @VisibleForTesting
+  protected S3DestinationStrictEncrypt(final S3DestinationConfigFactory configFactory) {
+    super(configFactory);
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(final JsonNode config) {
+    final S3DestinationConfig destinationConfig = this.configFactory.getS3DestinationConfig(config, super.storageProvider());
+
+    if (!S3BaseChecks.testCustomEndpointSecured(destinationConfig.getEndpoint())) {
+      return new AirbyteConnectionStatus()
```

Comment:
nitpick: add a comment explaining why we do an early return instead of returning the list of this status + `super.check(config)`